File: control-room/SosRelay/EventTypes/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SOS.EventTypes
{
    public class Game
    {
        public string arena { get; set; } = string.Empty;
        public Ball ball { get; set; } = new Ball();
        public bool hasTarget { get; set; } = false;
        public bool hasWinner { get; set; } = false;
        public bool isOT { get; set; } = false;
        public bool isReplay { get; set; } = false;
        public bool isSeries { get; set; } = false;
        public string localplayer { get; set; } = string.Empty;
        public int seriesLength { get; set; } = 0;
        public string target { get; set; } = string.Empty;
        public List<Team> teams { get; set; } = new List<Team>();
        public double time_milliseconds { get; set; } = 0.0;
        public int time_seconds { get; set; } = 0;
        public string winner { get; set; } = string.Empty;
    }
}

File: overlay-app/src/types/game.ts
import { Player } from './player';

export interface GameTeam {
  color_primary: string;
  color_secondary: string;
  name: string;
  score: number;
  logo: string;
}

export interface Game {
  arena: string;
  ball: {
    location: {
      X: number;
      Y: number;
      Z: number;
    };
    speed: number;
    team: number;
  };
  hasTarget: boolean;
  hasWinner: boolean;
  isOT: boolean;
  isReplay: boolean;
  isSeries: boolean;
  localplayer: string;
  seriesLength: number;
  target: string;
  teams: [GameTeam, GameTeam];
  time_milliseconds: number;
  time_seconds: number;
  winner: string;
}

export interface GameStateData {
  game: Game;
  players: Record<string, Player>; // TODO: this is a guess
}
